File: VB/Q321057MultipleGrids/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid

Namespace Q321057MultipleGrids
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			gridControl1.DataSource = FillDataTable()
			gridControl2.DataSource = FillDataTable()
		End Sub

		Public Shared Function FillDataTable() As DataTable
			Dim _dataTable As New DataTable()
			_dataTable.Columns.Add(New DataColumn("ID", GetType(Integer)))
			_dataTable.Columns.Add(New DataColumn("Name", GetType(String)))
			_dataTable.Rows.Add(New Object() { 1, "Apple" })
			_dataTable.Rows.Add(New Object() { 2, "Peach" })
			_dataTable.Rows.Add(New Object() { 3, "Banana" })
			Return _dataTable
		End Function
	End Class
End Namespace

File: CS/Q321057MultipleGrids/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;

namespace Q321057MultipleGrids
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            gridControl1.DataSource = FillDataTable();
            gridControl2.DataSource = FillDataTable();
        }

        public static DataTable FillDataTable()
        {
            DataTable _dataTable = new DataTable();
            _dataTable.Columns.Add(new DataColumn("ID", typeof(int)));
            _dataTable.Columns.Add(new DataColumn("Name", typeof(string)));
            _dataTable.Rows.Add(new object[] { 1, "Apple" });
            _dataTable.Rows.Add(new object[] { 2, "Peach" });
            _dataTable.Rows.Add(new object[] { 3, "Banana" });
            return _dataTable;
        }
    }
}
